File: server/DocumentsKM/Models/LinkedDoc.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Ссылочный документ (спр.)
    public class LinkedDoc
    {
        [Key]
        public Int16 Id { get; set; }

        // Шифр
        [Required]
        [MaxLength(4)]
        public string Code { get; set; }

        // Вид
        [Required]
        [ForeignKey("TypeId")]
        public virtual LinkedDocType Type { get; set; }

        // Обозначение
        [Required]
        [MaxLength(40)]
        public string Designation { get; set; }

        // Наименование
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }
    }
}

File: client/src/model/LinkedDoc.ts
import LinkedDocType from './LinkedDocType'

interface ILinkedDoc {
	id: number
	code: string
	name: string
	type: LinkedDocType
	designation: string
}

class LinkedDoc {
	id: number
	code: string
	name: string
	type: LinkedDocType
	designation: string

	constructor(obj?: ILinkedDoc) {
		this.id = (obj && obj.id) || 0
		this.code = (obj && obj.code) || ''
		this.name = (obj && obj.name) || ''
		this.type = (obj && obj.type) || null
		this.designation = (obj && obj.designation) || ''
	}
}

export default LinkedDoc
